pkg/kubectl/resource/builder.go

```diff
@@ -697,7 +697,7 @@ func (b *Builder) resourceMappings() ([]*meta.RESTMapping, error) {
 
 func (b *Builder) resourceTupleMappings() (map[string]*meta.RESTMapping, error) {
 	mappings := make(map[string]*meta.RESTMapping)
-	canonical := make(map[string]struct{})
+	canonical := make(map[schema.GroupVersionResource]struct{})
```

Comment:
previously, the returned map's key was the unqualified resource. this changes it to be the stringified fully-qualified resource (which is a human-readable format, not intended to be machine-readable). It looks like there's only a single call-site, which discards the key. Can this return a list instead, to avoid anything starting to depend on the key format?